test/firestore_test.dart

```diff
@@ -47,6 +47,14 @@ Future main() async {
     expect(first.id, isNot(second.id));
   });
 
+  test('Get documents via collection query', () async {
+    var query = await firestore
+        .collection('test')
+        .whereArrayContains('test_field', 'test_value')
```

Comment:
Should we write unit tests for all of these operators to make sure they work properly?

Reply:
Issue is you need a test firebase project. I'm not going to be the one to make that.
Also sensitive data and all that can't be pushed to the repo.

Reply:
Not saying we have to do this but you could mock out `runQuery()` and just confirm the correct query is being built. And then to test `runQuery()`, mock the client. 

@cachapa Do we have a mocking framework in the project? Something like [mockito](https://pub.dev/packages/mockito), maybe?

Again, code looks straight forward so I don't think we need tests here but just a nice to have.

Reply:
The project doesn't use mockito, no.
You're free to make a PR request for that though!